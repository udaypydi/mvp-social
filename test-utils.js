// test-utils.js
import { render, queries } from '@testing-library/react'

const customRender = (ui, options) =>
  render(ui, { queries: { ...queries }, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
