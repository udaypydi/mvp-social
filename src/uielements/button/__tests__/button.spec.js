import React from 'react';
import { render } from 'test-utils';
import  Button from 'uielements/button/button.component.jsx';

it('should take a snapshot', () => {
    const { asFragment } = render(<Button buttonText="test button" />)
    
    expect(asFragment(<Button buttonText="test button" />)).toMatchSnapshot()
});
