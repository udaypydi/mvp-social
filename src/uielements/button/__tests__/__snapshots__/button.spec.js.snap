// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <button
    class="bg-transparent hover:bg-blue-500 text-indigo-100 font-semibold hover:text-white py-2 px-4 border border-indigo-100 hover:border-transparent rounded"
    type="button"
  >
    test button
  </button>
</DocumentFragment>
`;
